src/create-menu/utils/change-menu-item-states-by-id.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const { Menu, MenuItem } = require('electron')
+
+const MenuIpcChannelHandlers = require(
+  '../../window-creators/main-renderer-ipc-bridge/menu-ipc-channel-handlers'
+)
+const wins = require('../../window-creators/windows')
+const WINDOW_NAMES = require('../../window-creators/window.names')
+const isMainWinAvailable = require(
+  '../../helpers/is-main-win-available'
+)
+
+module.exports = (menuItemId, opts) => {
```

Comment:
I would suggest always accepting an array `([{ menuItemId, opts }, ...], fallbackOpts)` in this method since we create an array from a single item anyway. IMHO, parameters list is misleading otherwise.

Reply:
I'm not against strict typing, but I'd like to keep some flexibility here as it is
Since this is pure JS, we should rely on the `Defensive Programming`, and this is an old and reliable way of guaranteeing an array in this place

Reply:
I am up for different styles and paradigms. I'm this particular case I'm missing the purpose of m multiple types accepted since it always creates an array anyway. At the same time, there is only one place where method is invoked with an array and it could be easily modified to have multiple calls instead. Additionally, `opts` logic with the current approach is not straightforward and serves multiple purposes simultaneously, which is not ideal.
After all, to me the current implementation looks unnecessarily overcomplicated, which translates to maintenance problems eventually